src/error/GraphQLError.ts

```diff
@@ -76,15 +113,16 @@ export class GraphQLError extends Error {
    */
   readonly extensions: GraphQLErrorExtensions;
 
-  constructor(
-    message: string,
-    nodes?: ReadonlyArray<ASTNode> | ASTNode | null,
-    source?: Maybe<Source>,
-    positions?: Maybe<ReadonlyArray<number>>,
-    path?: Maybe<ReadonlyArray<string | number>>,
-    originalError?: Maybe<Error & { readonly extensions?: unknown }>,
-    extensions?: Maybe<GraphQLErrorExtensions>,
-  ) {
+  constructor(...rawArgs: BackwardsCompatibleArgs) {
+    const {
+      message,
+      nodes,
+      source,
+      positions,
+      path,
+      originalError,
+      extensions,
+    } = toNormalizedArgs(rawArgs);
```

Comment:
Current form is bad for doc generation and DX.
Please use TS overloading with two separate signatures.

Reply:
@IvanGoncharov Done.